ImperatorToCK3/Source/Mappers/RegionMapper/CK3RegionMapper.cpp

```diff
@@ -0,0 +1,177 @@
+#include "CK3RegionMapper.h"
+#include "Log.h"
+#include "OSCompatibilityLayer.h"
+#include <filesystem>
+#include <fstream>
+#include "ParserHelpers.h"
+
+namespace fs = std::filesystem;
+
+void mappers::CK3RegionMapper::loadRegions(const std::string& ck3Path, CK3::LandedTitles& landedTitles)
+{
+	LOG(LogLevel::Info) << "-> Initializing Geography";
+	
+	auto regionFilename = ck3Path + "/game/map_data/geographical_region.txt";
+	auto islandRegionFilename = ck3Path + "/game/map_data/island_region.txt";
+	
+	std::ifstream regionStream(fs::u8path(regionFilename));
+	if (!regionStream.is_open())
+		throw std::runtime_error("Could not open game/map_data/geographical_region.txt!");
+
+	std::ifstream islandRegionStream(fs::u8path(islandRegionFilename));
+	if (!islandRegionStream.is_open())
+		throw std::runtime_error("Could not open game/map_data/island_region.txt!");
+	
+	loadRegions(landedTitles, regionStream, islandRegionStream);
+
+	regionStream.close();
+	islandRegionStream.close();
+}
+
+void mappers::CK3RegionMapper::loadRegions(CK3::LandedTitles& landedTitles, std::istream& regionStream, std::istream& islandRegionStream)
+{
+	registerRegionKeys();
+	parseStream(regionStream);
+	clearRegisteredKeywords();
+
+	registerRegionKeys();
+	parseStream(islandRegionStream);
+	clearRegisteredKeywords();
+	
+	for (const auto& [titleName, title] : landedTitles.getTitles())
```

Comment:
unless this part can't be done twice, of course.

Reply:
It can, but doing so would be redundant. I think it's better to keep separate streams instead of two function calls. If game files are OK, there should always be these two region files, and it's cleaner to call once.

Reply:
Not an issue at all. whatever is more efficient and serves the purpose.